mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -744,7 +744,17 @@ fn dex_fee_rate(base: &str, rel: &str) -> MmNumber {
 
 pub fn dex_fee_amount(base: &str, rel: &str, trade_amount: &MmNumber, min_tx_amount: &MmNumber) -> MmNumber {
     let rate = dex_fee_rate(base, rel);
-    let fee_amount = trade_amount * &rate;
+
+    let fee_amount = if base == "KMD" {
+        // Drop the fee by 25%, which will be burned during the taker fee payment.
+        //
+        // This cut will be dropped before return if the final amount is less than
+        // the minimum transaction amount.
+        (trade_amount * &rate) * MmNumber::from("0.75")
+    } else {
+        trade_amount * &rate
+    };
+
     if &fee_amount < min_tx_amount {
         min_tx_amount.clone()
```

Comment:
There is a problem in the calculation here, let's say minimum transaction amount is 1000 sats and the fee amount (total fee before burning) is 1100, then (1100 * 0.75) = 825 so the taker pays 1000 because `825 < 1000` when they should pay 1100 after we check if we will burn some KMD or not and it will be not.

Reply:
P.S. I think we should leave the `dex_fee_amount` function as it were before and this will be for the total fee, and this calculation https://github.com/KomodoPlatform/komodo-defi-framework/blob/38ae93c6143e19fa58a28963eb025678aa9b3580/mm2src/coins/utxo/utxo_common.rs#L1675 should be done only in one place preferably inside `should_taker_burn_kmd` function and never outside it where the function returns the actual dex fee amount and the burn amount, if burn amount is 0 (or None if you prefer to make it an option) we only add one output.

Reply:
> There is a problem in the calculation here

Yeah, right. Seems like for KMD we always return the minimum transaction unless it's (trade_amount * &rate) * MmNumber::from("0.75") > minimum_transaction. Will fix that.

> I think we should leave the dex_fee_amount function as it were before and this will be for the total fee, and this calculation

This is the central function which taker and maker uses often. If we don't update this function, then we need to spreat `should_kmd...` function call all over the swap functions. Otherwise swaps will fail.

Reply:
> This is the central function which taker and maker uses often. If we don't update this function, then we need to spreat should_kmd... function call all over the swap functions. Otherwise swaps will fail.

How about making `dex_fee_amount` return an enum, e.g.
```rust
enum TakerFee {
    // A fee output that is sent entirely to the dex address
    DexOnly(MmNumber),
    // A fee output that is split between burning and sending to the fee address
    DexAndBurn {
        dex_fee: MmNumber,
        burn_fee: MmNumber,
    },
}
```
This will help you to not miss any instances that need modifying, e.g. the miner fee to send the taker fee is calculated incorrectly where only one output is considered
https://github.com/KomodoPlatform/komodo-defi-framework/blob/8d0a58356a4806f88e695e5541ff870daffddba8/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L2358-L2361
https://github.com/KomodoPlatform/komodo-defi-framework/blob/38ae93c6143e19fa58a28963eb025678aa9b3580/mm2src/coins/utxo/utxo_common.rs#L4080-L4087

Reply:
Enum is def. better solution. I will implement it that way before Monday.